src/main/java/uk/ac/ic/wlgitbridge/bridge/lock/InProcessLockFactory.java

```diff
@@ -0,0 +1,37 @@
+package uk.ac.ic.wlgitbridge.bridge.lock;
+
+import uk.ac.ic.wlgitbridge.util.Log;
+
+import java.util.HashMap;
+
+public final class InProcessLockFactory implements ProjectLockFactory {
+  private final HashMap<String, InProcessLock> locks;
+  private final OperationTracker operationTracker;
+
+  public InProcessLockFactory() {
+    // TODO: Note for review: I'm not sure we need to keep these in a map,
+    //       either that, or we need to clean them up when the lock is released?
+    //       or, these need to be reference counted?
```

Comment:
Yeah, I think it would be better to reference count. Sorry that I didn't think of this in our pairing sessions. A `HashMap<String, Pair<InProcessLock, int>>` should do it?

I am assuming the existence of `Pair` here, coming from a C++ background. ykwim anyway I think.